servers/visual/visual_server_scene.cpp

```diff
@@ -2294,20 +2294,19 @@ _FORCE_INLINE_ static void _light_capture_sample_octree(const RasterizerStorage:
 				half >>= 1;
 			}
 
-			if (cell == RasterizerStorage::LightmapCaptureOctree::CHILD_EMPTY) {
-				alpha[c][n] = 0;
-			} else {
+			if (cell != RasterizerStorage::LightmapCaptureOctree::CHILD_EMPTY) {
 				alpha[c][n] = p_octree[cell].alpha;
 
 				for (int i = 0; i < 6; i++) {
 					//anisotropic read light
 					float amount = p_dir.dot(aniso_normal[i]);
-					if (amount < 0) {
-						amount = 0;
+
+					if (amount > 0) {
+						constexpr float ONE_1024TH = 1.0 / 1024.0;
+						color[c][n].x += p_octree[cell].light[i][0] * ONE_1024TH * amount;
+						color[c][n].y += p_octree[cell].light[i][1] * ONE_1024TH * amount;
+						color[c][n].z += p_octree[cell].light[i][2] * ONE_1024TH * amount;
```

Comment:
These might give marginally different answers to the division but I'm guessing it doesn't matter (and the compiler should optimize this to a single multiplier).

Reply:
> (and the compiler should optimize this to a single multiplier).

This is only the case with `-ffast-math`, which we don't use. This is because the result isn't guaranteed to always be the same.

Reply:
Based on my comment here https://github.com/godotengine/godot/pull/80764#discussion_r1298710909 these changes make 9% improvement.
Your comments confuse me, do you want I revert them because of floating-point precision errors? I think floating-point precision errors doesn't important here because this isn't a code in physics engine and these values interpolated later and final result isn't noticeable to naked eye.

Reply:
Sorry I wasn't specific, I was meaning it is more likely to get optimized in the case of the two multiplies (i.e. after the PR, rather than before). I don't think they need reverting (as this gives the speedup), this is just to note there may be a slight difference in the calculated result, which hopefully does not matter in this case, as this is just a color.

Reply:
Thanks for confirmation.